core/controllers/reader.py

```diff
@@ -55,6 +55,17 @@
 def require_playable(handler):
     """Decorator that checks if the user can play the given exploration."""
     def test_can_play(self, exploration_id, **kwargs):
+        """
+        Args:
+            exploration_id: Id of the exploration
+            kwargs: Any other keyworded args.
+
+        Returns:
+            either None if exploration_id is in disabled ids or
```

Comment:
handler or None. If the exploration is disabled, the disabled_exploration.html page is rendered. Otherwise, if the exploration can be played by the user, the corresponding handler is returned.